### ISSUE TITLE
[Bug] [macOS] MainPage=NavigationPage causes crash with version 4.3 on High Sierra

### ISSUE THREAD

**Issue by zhuxinghan** [xamarin/Xamarin.Forms#8261]
### Description

After updating Xamarin.Forms to version 4.3, I found my apps that use NavigationPage as MainPage can't launch (crashes before window is shown) on macOS 10.13 High Sierra, but there're no problems on macOS 10.15 Catalina. Xamarin.Forms 4.2 is OK for both High Sierra and Catalina.

### Basic Information

- Version with issue: the first version of 4.3
- Last known good version: the last version of 4.2

### Reproduction Link

A simple project with MainPage = new NavigationPage(...) is able to reproduce.

You can use the sample from my another issue ( #7859 ) for convenience. Run it on macOS High Sierra (note to update Xamarin.Forms to 4.3 and set the LSMinimumSystemVersion to 10.13 before running), you can find that it crashes before the window is shown. Then you can replace the NavigationPage with a normal page in App.xaml.cs, and the app can launch successfully.

**Comment by ErwinDeGroot** [comment]
I have the same issue. I believe it is because the NativeToolbarTracker (used in the NavigationPage somewhere). In the ConfigureToolbar method, the 'CenteredItemIdentifier' is set, but this property is only supported on Mojave and will crash on older OS versions.

**Comment by jfversluis** [comment]
@ErwinDeGroot would you be able to verify that yourself by running the Xamarin.Forms repo? Or, if I put in the fix, generate a NuGet from that verify that it actually fixes the issue? 

I don't have an older Mac OS version to test on anymore unfortunately.

**Comment by jfversluis** [comment]
You, or anyone reading this, should be able to get NuGets from the associated PR build and try it out and please let us know :)
